From 04f32107fba7ea7c186acf017970215af5de3297 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Thu, 27 Nov 2025 20:07:59 +0800
Subject: [PATCH 43/63] loonggpu: ensure clearing the mapping field of ttm
 pages

As the name of "lg_ttm_unmap_and_unpopulate_pages" suggests, it should
unmap them.

Following the reference implementation change.

Link: https://git.kernel.org/torvalds/c/21856e1e3425
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 include/loonggpu_helper.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/loonggpu_helper.h b/include/loonggpu_helper.h
index 6340701..c6dd605 100644
--- a/include/loonggpu_helper.h
+++ b/include/loonggpu_helper.h
@@ -1062,6 +1062,8 @@ static inline void lg_ttm_unmap_and_unpopulate_pages(struct loonggpu_device *ade
 #if defined(LG_DRM_TTM_TTM_PAGE_ALLOC_H_PRESENT)
 	ttm_unmap_and_unpopulate_pages(adev->dev, ttm);
 #else
+	for (pgoff_t i = 0; i < ttm->num_pages; i++)
+		ttm->pages[i]->mapping = NULL;
 	ttm_pool_free(&adev->mman.bdev.pool, ttm);
 #endif
 }
-- 
2.52.0

